tensorflow/python/keras/engine/training.py

```diff
@@ -543,8 +543,11 @@ def compile(self,
       self.optimizer = self._get_optimizer(optimizer)
       self.compiled_loss = compile_utils.LossesContainer(
           loss, loss_weights, output_names=self.output_names)
-      self.compiled_metrics = compile_utils.MetricsContainer(
-          metrics, weighted_metrics, output_names=self.output_names)
+      mc = compile_utils.MetricsContainer(metrics,
+                                          weighted_metrics,
+                                          loss=loss,
```

Comment:
Given my other comment above, maybe this should be using the compiled/normalized `self.compiled_loss` value(s) rather than the generic `loss` one that can be in 'any' form? I don't know what a `LossesContainer` contains in particular, but I'm comparing to the TF 2.1 code, where there didn't need to be a special case for string forms like `"bce"`:

https://github.com/tensorflow/tensorflow/blob/3ffdb91f122f556a74a6e1efd2469bfe1063cb5c/tensorflow/python/keras/engine/training_utils.py#L1114-L1121

Reply:
OK I modified the patch using the compiled loss. Can you check again?

Reply:
I reran the https://colab.research.google.com/gist/huonw/4ac8796f598bc742b476fd8e36a1f866/gcn-link-prediction.ipynb notebook and it seemed to work (other than the change from `model.compiled_metrics._loss` to `model.compiled_metrics._loss_container`). The cell that tests various ways to specify binary crossentropy now has output:

```
'bce' binary_accuracy
'binary_crossentropy' binary_accuracy
<function binary_crossentropy at 0x7f94729b1950> binary_accuracy
<tensorflow.python.keras.losses.BinaryCrossentropy object at 0x7f94629c13c8> binary_accuracy
<function binary_crossentropy at 0x7f94729b1950> binary_accuracy
```

And the accuracy and validation accuracy metrics now look correct too 👍

Reply:
Thanks for testing